client/my-sites/hosting-dashboard-opt-in-banner/index.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+	Button,
+	Card,
+	CardBody,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import TrackComponentView from 'calypso/lib/analytics/track-component-view';
+import { useDispatch, useSelector } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice } from 'calypso/state/notices/actions';
+import { savePreference } from 'calypso/state/preferences/actions';
+import {
+	getPreference,
+	isFetchingPreferences,
+	isSavingPreference,
+	preferencesLastSaveError,
+} from 'calypso/state/preferences/selectors';
+import illustratioUrl from './illustration.svg';
+import type { HostingDashboardOptIn } from '@automattic/api-core';
+
+export default function HostingDashboardOptInBanner() {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const savedPreference = useSelector(
+		( state ) => getPreference( state, 'hosting-dashboard-opt-in' ) as HostingDashboardOptIn | null
+	);
+
+	const isFetching = useSelector( isFetchingPreferences );
+	const isSaving = useSelector( isSavingPreference );
+	const lastSaveError = useSelector( preferencesLastSaveError );
+
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+
+	const handleClick = async () => {
+		setIsSubmitting( true );
+
+		recordTracksEvent( 'calypso_dashboard_opt_in_banner_click' );
+
+		const preference = {
+			value: 'opt-in',
+			updated_at: new Date().toISOString(),
+		} satisfies HostingDashboardOptIn;
+
+		await dispatch( savePreference( 'hosting-dashboard-opt-in', preference ) );
+
+		if ( lastSaveError ) {
+			dispatch(
+				errorNotice( translate( 'Failed to save preference.' ), {
+					duration: 5000,
+				} )
+			);
+		} else {
+			window.location.href = '/v2';
+		}
+	};
+
+	if ( isFetching || savedPreference?.value === 'opt-out' ) {
```

Comment:
What if `savedPreference.value === 'opt-in'`? Currently if you navigate directly to `/sites` after opting in you see the same banner again. I wasn't expecting that. I was expecting the banner to be hidden after I'd opt'd in. Although we could show a different banner? Maybe one that says "Looking for the new hosting dashboard?"

Reply:
> I wasn't expecting that. I was expecting the banner to be hidden after I'd opt'd in

The idea is that the user might get lost in the old dashboard without knowing how to get the new one. E.g. when they have bookmarked `/sites`. So we need to guide them there.

> Although we could show a different banner? Maybe one that says "Looking for the new hosting dashboard?"

Yeah, I think we should show something like this... let me come up with something.

Reply:
Updated the copy to be like this when the user has already opted in but for some reason visits the old dashboard. Let me know what you think. Also cc @lucasmendes-design 

<img width="268" height="274" alt="Image" src="https://github.com/user-attachments/assets/3ac0e737-c3de-4c15-a786-8eb89302a0b0" />